sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts

```diff
@@ -229,5 +238,59 @@ describe(
         currentMSFAA.id,
       );
     });
+
+    it("Should cancel a pending MSFAA record when the same MSFAA record is received in the response file and persist a MSFAA cancellation notification for the same.", async () => {
+      // Arrange
+      // Create a MSFAA record in pending state with the same MSFAA number as in the msfaa-full-time-receive-file-with-single-cancellation-record.dat
+      const student = await saveFakeStudent(db.dataSource);
+      const studentUserId = student.user.id;
+      const notificationMessageType = NotificationMessageType.MSFAACancellation;
+      await db.msfaaNumber.save(
+        createFakeMSFAANumber(
+          { student },
+          {
+            msfaaState: MSFAAStates.Pending,
+            msfaaInitialValues: {
+              msfaaNumber: FULL_TIME_SAMPLE_MSFAA_NUMBER,
+              offeringIntensity: OfferingIntensity.fullTime,
+            },
+          },
+        ),
+      );
+      // Queued job.
+      const job = createMock<Job<void>>();
+      mockDownloadFiles(sftpClientMock, [
+        MSFAA_FULL_TIME_RECEIVE_FILE_WITH_SINGLE_CANCELLATION_RECORD,
+      ]);
+      // Act
+      // Now cancel the MSFAA record.
+      const processResult = await processor.processMSFAA(job);
+      // Assert
+      expect(processResult).toStrictEqual([
+        {
+          processSummary: [
+            `Processing file ${MSFAA_FULL_TIME_RECEIVE_FILE_WITH_SINGLE_CANCELLATION_RECORD}.`,
+            "File contains:",
+            "Confirmed MSFAA records (type R): 0.",
+            "Cancelled MSFAA records (type C): 1.",
+            "Record from line 1, updated as cancelled.",
+          ],
+          errorsSummary: [],
+        },
+      ]);
+      // Get the notification using the student and notification message id. Since a fake student is created for every test run, it will ensure uniqueness.
+      // Expecting one notification for the cancelled MSFAA record.
+      const notificationsQuery = db.notification
```

Comment:
can't we use Typeform query

Reply:
I tried to use it at first but then struggled in putting up the where clause using the typeorm query.

Reply:
Example:
```js
   {
      select: {
        id: true,
        overawardNotes: { noteType: true, description: true },
      },
      relations: { overawardNotes: true },
      where: { id: overawardResponse.body.id },
    }
```

Reply:
Thanks @andrepestana-aot